platform/posix/transport/src/transport_reconnect_posix.c

```diff
@@ -38,32 +38,32 @@ bool Transport_ReconnectBackoffAndSleep( TransportReconnectParams_t * pReconnect
     uint32_t jitter = 0;
 
     /* If MAX_RECONNECT_ATTEMPTS is set to 0, try forever */
-    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPS ) ||
-        ( 0 == MAX_RECONNECT_ATTEMPS ) )
+    if( ( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPTS ) &&
+          ( pReconnectParams->nextBackOffSec <= MAX_RECONNECT_BACKOFF_SECONDS ) ) ||
```

Comment:
So what is purpose of adding ( pReconnectParams->nextBackOffSec <= MAX_RECONNECT_BACKOFF_SECONDS ) ? If  nextBackOffSec > MAX_RECONNECT_BACKOFF_SECONDS , but attempts are < MAX_RECONNECT_ATTEMPTS , application should still sleep(). 
By this change, it will skip sleeping and instead return error.

Reply:
Good point there. Have updated the logic to move the check to be nested within the outer if-block.